### DISCUSSION TITLE
Does `wezterm` expose any functionality to check if a pane exists in given direction?

### DISCUSSION THREAD

**Discussion by schardev** [wezterm/wezterm#2990] [Q&A] [answered]
Like I want to check if a pane exists in the direction, if it does I'll move to the pane else I'll send the key to the terminal instead (since wezterm will swallow the key otherwise).

So, if I press <kbd>Ctrl</kbd> + <kbd>→</kbd> then move to the pane in right direction **if it exists** else send that key to ternimal since it is mapped to complete partial shell command completion.

**Comment by wez** [comment]
No, wezterm doesn't have that functionality

**Comment by distek** [comment, marked as answer]
Would this work?

```lua
	{
		key = "RightArrow",
		mods = "CTRL",
		action = wezterm.action_callback(function(win, pane)
			local tab = pane:tab()

			if tab:get_pane_direction("Right") ~= nil then
				win:perform_action(act.ActivatePaneDirection("Right"), pane)
				return
			end
			act.SendKey({ key = "RightArrow", mods = "CTRL" })
		end),
	},
```

The docs for `SendKey` state:
> Note that this rebinding effect only applies to the input that is about to be sent to the pane; **it doesn't get re-evaluated against the key assignments you've configured in wezterm again.**

So I _believe_ this should work

**Reply by schardev** [reply]
I think this is *exactly* what I asked for. Since it was [added](https://github.com/wez/wezterm/commit/7221c1b3f3165f05995bf100b2f2e7181e514f11) very recently I might have to wait for it to appear in arch repos. Thanks for the ping :)

**Reply by distek** [reply]
Ah! Forgot I was on the AURs git package lol
